aten/src/ATen/core/Type.h

```diff
@@ -61,6 +61,7 @@ enum class TypeID {
   SparseCPUInt,
   SparseCPULong,
   SparseCPUShort,
+  MkldnnCPUFloat,
```

Comment:
what's the plan to add the full list of types?

Reply:
MKL-DNN currently only supports fp32 and there might be more types supported but I don't think it would cover the full list. Do you have particular concern in mind?

Reply:
The issue is that we are trying to move away from dispatch being (Backend + ScalarType).  CC @li-roy.

Reply:
i think this is fine. if we dispatch off Backend only, each Backend can choose whatever it wants to support in regards to ScalarType. i.e. MLK-DNN can just support float